Patch-Source: https://github.com/php/php-src/commit/a2c5b4ec05da0c26370332988b30befa43e29e77
From a2c5b4ec05da0c26370332988b30befa43e29e77 Mon Sep 17 00:00:00 2001
From: Niels Dossche <7771979+nielsdos@users.noreply.github.com>
Date: Thu, 23 May 2024 22:20:37 +0200
Subject: [PATCH] Fix GH-14307: Test curl_basic_024 fails with curl 8.8.0

Curl changed the behaviour, from the changelog:
  - lib: make protocol handlers store scheme name lowercase curl/curl@c294f9c

From the docs: "The returned scheme might be upper or lowercase. Do
comparisons case insensitively."

Closes GH-14312.
---
 NEWS                               | 3 +++
 ext/curl/tests/curl_basic_024.phpt | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/ext/curl/tests/curl_basic_024.phpt b/ext/curl/tests/curl_basic_024.phpt
index b16bfed03835f..84216bde308eb 100644
--- a/ext/curl/tests/curl_basic_024.phpt
+++ b/ext/curl/tests/curl_basic_024.phpt
@@ -25,7 +25,7 @@ var_dump(0 === curl_getinfo($ch, CURLINFO_PROXY_SSL_VERIFYRESULT));
 var_dump(curl_getinfo($ch, CURLINFO_SCHEME));
 curl_close($ch);
 ?>
---EXPECT--
+--EXPECTF--
 bool(true)
 bool(true)
-string(4) "HTTP"
+string(4) "%r(HTTP|http)%r"
